lua/entities/gmod_tardis/modules/sh_health.lua

```diff
@@ -174,6 +254,57 @@ if SERVER then
 		self:FlashLight(1.5)
 	end
 
+	ENT:AddHook("Think", "redecorate-matnew", function(self)
+		local demattime = self:GetData("redecorate-demattime")
+		if demattime and CurTime() - demattime > 0.5 then
+			self:SetData("redecorate-demattime", nil)
+
+			local ent = self:GetData("redecorate_next")
+			if IsValid(ent) then
+				local pos = self:GetPos() + Vector(0,0,0.02)
+				local ang = self:GetAngles()
+				ent:SetPos(pos)
+				ent:SetAngles(ang)
+				-- this is repeated in case it was moved at first second
+
+				ent:SetCollisionGroup(COLLISION_GROUP_WORLD)
+				self:SetParent(ent)
+				self:SetCollisionGroup(COLLISION_GROUP_IN_VEHICLE)
+				-- you can't interact with both at once
+
+				ent:SetFastRemat(true)
+				ent:Mat()
+			end
+		end
+	end)
+
+	ENT:AddHook("StopDemat", "redecorate-dissapear", function(self)
+		if self:GetData("redecorate") then
+			self:Remove()
+		end
+	end)
+
+	ENT:AddHook("StopMat", "redecorate_restore_data", function(self)
+		if self:GetData("parent-saved-data") then
+
+			local saved_data = self:GetData("parent-saved-data")
+
+			for k,v in ipairs(migrated_data_names) do
+				self:SetData(v, saved_data[v], true)
+			end
+			if IsValid(self.interior) then
+				self.interior:SetData("security", self:GetData("parent-security", false), true)
```

Comment:
Would it make more sense to have an interior_migrated_data_names or something here for this?